sql/core/src/test/scala/org/apache/spark/sql/execution/QueryExecutionSuite.scala

```diff
@@ -220,7 +220,8 @@ class QueryExecutionSuite extends SharedSparkSession {
     assertNoTag(tag5, df.queryExecution.sparkPlan)
   }
 
-  test("Logging plan changes for execution") {
+  // TODO(SPARK-44074): re-enable this test after SPARK-44074 resolved
+  ignore("Logging plan changes for execution") {
```

Comment:
After this PR, we need to backport the changes of SPARK-44074 to branch-3.4 to re-enable this case.

Can the changes of SPARK-44074 be merged into the current PR?

For the branch-3.3, I will submit pr later. Afterwards, the `build_and_test.yml` of the master branch also needs to be fixed to let the daily test use these tags.

Reply:
cc @HyukjinKwon @sunchao @dongjoon-hyun

Reply:
On the other hand, does branch-3.4 not need SPARK-4405 (https://github.com/apache/spark/pull/41601/files)?

Reply:
> Can the changes of [SPARK-44074](https://issues.apache.org/jira/browse/SPARK-44074) be merged into the current PR?

@LuciferYang OK for me, we can also just cherry-pick it separately after this PR. Either way is fine IMO.

Reply:
> On the other hand, does branch-3.4 not need [SPARK-4405](https://issues.apache.org/jira/browse/SPARK-4405) (https://github.com/apache/spark/pull/41601/files)?

Looks like this will be useful to branch-3.4 too.

Reply:
@dongjoon-hyun Would you mind backporting SPARK-44057 first? If it's inconvenient, I can handle it.